File: server/src/main/java/com/xcore/application/authentication/controllers/AuthController.java
package com.xcore.application.authentication.controllers;

import com.xcore.application.authentication.controllers.api.CurrentAuthInfoResponse;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping("/api/auth")
@RestController
public class AuthController {

    @GetMapping("/info")
    public CurrentAuthInfoResponse getCurrentAuthInfo() {

        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

        return new CurrentAuthInfoResponse(authentication);
    }

    @PostMapping("/logout")
    public CurrentAuthInfoResponse logout() {

        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        authentication.setAuthenticated(false);

        return new CurrentAuthInfoResponse(authentication);
    }

}

File: server/src/main/scala/com/xcore/application/authentication/controllers/AuthController.scala
package com.xcore.application.authentication.controllers;

import com.xcore.application.authentication.controllers.api.{CurrentAuthInfoResponse, SignUpRequest, SignUpResponse};
import com.xcore.application.authentication.models.user.IAppUserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.{GetMapping, PostMapping, RequestMapping, RestController};

@RestController
@RequestMapping(Array ("/api/auth") )
class AuthController {

  @Autowired
  private var appUserRepository: IAppUserRepository = _;

  @GetMapping(Array("/info"))
  def getCurrentAuthInfo: CurrentAuthInfoResponse = {
    val authentication = SecurityContextHolder.getContext.getAuthentication;

    new CurrentAuthInfoResponse(authentication);
  }

  @PostMapping(Array("/logout"))
  def logout: CurrentAuthInfoResponse = {
    val authentication = SecurityContextHolder.getContext.getAuthentication;
    authentication.setAuthenticated(false);

    new CurrentAuthInfoResponse(authentication);
  }

  @PostMapping(Array("/sign-up"))
  def signUp(request: SignUpRequest): SignUpResponse = {


    new SignUpResponse();
  }

}